pkg/distributor/distributor.go

```diff
@@ -818,12 +819,19 @@ func (d *Distributor) prePushRelabelMiddleware(next push.Func) push.Func {
 			ts := req.Timeseries[tsIdx]
 
 			if mrc := d.limits.MetricRelabelConfigs(userID); len(mrc) > 0 {
-				l, keep := relabel.Process(mimirpb.FromLabelAdaptersToLabels(ts.Labels), mrc...)
+				lb := labels.NewBuilder(mimirpb.FromLabelAdaptersToLabels(ts.Labels))
+				lb.Set(model.MetaLabelPrefix+"tenant_id", userID)
```

Comment:
Let's put `model.MetaLabelPrefix+"tenant_id"` into a constant, to make sure it's not recomputed on every push request and timeseries. Maybe compiler is already smart to do that, but we can use the constant later when cleaning up. See next comment.